types/utxo.go

```diff
@@ -33,65 +45,92 @@ func NewBaseUTXO(addr common.Address, inputaddr [2]common.Address, amount uint64
 }
 
 //Implements UTXO
-func (baseutxo *BaseUTXO) GetAddress() common.Address {
-	return baseutxo.Address
+func (baseutxo BaseUTXO) GetAddress() []byte {
+	return baseutxo.Address.Bytes()
 }
 
 //Implements UTXO
-func (baseutxo *BaseUTXO) SetAddress(addr common.Address) error {
+func (baseutxo *BaseUTXO) SetAddress(addr []byte) error {
 	if !utils.ZeroAddress(baseutxo.Address) {
-		return errors.New("cannot override BaseUTXO Address")
+		return fmt.Errorf("address already set to: %X", baseutxo.Address)
 	}
-	if utils.ZeroAddress(addr) {
-		return errors.New("address provided is nil")
+	address := common.BytesToAddress(addr)
+	if utils.ZeroAddress(address) {
+		return fmt.Errorf("invalid address provided: %X", address)
 	}
-	baseutxo.Address = addr
+	baseutxo.Address = address
+	return nil
 }
 
 //Implements UTXO
 func (baseutxo *BaseUTXO) SetInputAddresses(addrs [2]common.Address) error {
 	if !utils.ZeroAddress(baseutxo.InputAddresses[0]) {
-		return errors.New("cannot override BaseUTXO Address")
+		return fmt.Errorf("input addresses already set to: %X, %X", baseutxo.InputAddresses[0], baseutxo.InputAddresses[1])
 	}
 	if utils.ZeroAddress(addrs[0]) {
-		return errors.New("address provided is nil")
+		return fmt.Errorf("invalid address provided: %X", addrs[0])
 	}
 	baseutxo.InputAddresses = addrs
+	return nil
 }
 
 //Implements UTXO
-func (baseutxo *BaseUTXO) GetInputAddresses() [2]common.Address {
+func (baseutxo BaseUTXO) GetInputAddresses() [2]common.Address {
 	return baseutxo.InputAddresses
 }
 
 //Implements UTXO
-func (baseutxo *BaseUTXO) GetAmount() uint64 {
+func (baseutxo BaseUTXO) GetAmount() uint64 {
 	return baseutxo.Amount
 }
 
 //Implements UTXO
 func (baseutxo *BaseUTXO) SetAmount(amount uint64) error {
 	if baseutxo.Amount != 0 {
-		return errors.New("cannot override BaseUTXO amount")
+		return fmt.Errorf("amount already set to: %d", baseutxo.Amount)
 	}
 	baseutxo.Amount = amount
+	return nil
 }
 
-func (baseutxo *BaseUTXO) GetPosition() utxo.Position {
+func (baseutxo BaseUTXO) GetPosition() utxo.Position {
 	return baseutxo.Position
 }
 
 func (baseutxo *BaseUTXO) SetPosition(position utxo.Position) error {
 	if baseutxo.Position.IsValid() {
-		return errors.New("cannot override BaseUTXO position")
+		return fmt.Errorf("position already set to: %v", baseutxo.Position)
+	} else if !position.IsValid() {
+		return errors.New("invalid position provided")
+	}
+
+	plasmaposition, ok := position.(PlasmaPosition)
+	if !ok {
+		return errors.New("position must be of type PlasmaPosition")
+	}
+	baseutxo.Position = plasmaposition
+	return nil
+}
+
+func (baseutxo BaseUTXO) GetDenom() string {
+	return "Ether"
+}
+
+func (baseutxo *BaseUTXO) SetDenom(denom string) error {
```

Comment:
SetDenom should throw error here

Reply:
handler in x/utxo needs to be able to set the Denom

Reply:
Sorry SetDenom should just do nothing, just return nil